rest/src/main/java/io/syndesis/rest/v1/util/Urls.java

```diff
@@ -17,17 +17,33 @@
 
 import java.net.URI;
 import java.net.URISyntaxException;
+import java.util.Optional;
 
 import javax.servlet.http.HttpServletRequest;
 
 public final class Urls {
 
+    /* default */ static final String ORIGIN_HEADER = "X-Forwarded-Origin";
+
     private Urls() {
         // Utility class, only static methods
     }
 
+    public static URI absoluteTo(final HttpServletRequest httpRequest, final URI url) {
```

Comment:
I don't think so URLs [are evil](https://docs.oracle.com/javase/8/docs/api/java/net/URL.html#equals-java.lang.Object-)

    Since hosts comparison requires name resolution, this operation is a blocking operation.